### PR TITLE
Add function to create run dev by module

### LINKED ISSUE #215: Create a add_run_dev_module
We should have a function that allows to create a run_dev for a specific module, called `add_run_dev_module`.

### PR DESCRIPTION
**Pull request by Cervangirard**
My deer Colin, see my contribution to {golem}.

A new function to add one run_dev by module.

One thing to notice, parameters for module have to be a string
close #215

### CODE REVIEW

File: R/create_run_dev.R [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+#' @examples
+#' 
+#' \dontrun{
+#' create_run_dev("ok", 
+#'                mod_ok_ui,
+#'                mod_ok_server, 
+#'                r= "reactiveValues(message = 'ok')", 
+#'                message = "'have to be string'")
+#' }
+create_run_dev <- function(name,
```
Comment (ColinFay) [inline R/create_run_dev.R]: Can you rename that add_run_dev_module so it's consistent with the rest of the naming convention?

Comment (Cervangirard) [reply R/create_run_dev.R]: Sorry to push directly into master, I forgot that we have a branch dev on golem :D. But i write this part on golem book.... tough life... :)

**Comment by ColinFay** [comment]
+ The name of the file should be run throught `basename()`

``` r
> golem::add_module("plop", open = FALSE)
● Couldn't find golem working directory
✔ Definining golem working directory as `.`
You can change golem working directory with set_golem_wd('path/to/wd')
✔ Definining golem working directory as `/Users/colin/Seafile/documents_colin/R/trash/ooooooo`
✔ File created at R/mod_plop.R
◼ Go to R/mod_plop.R
> golem::add_run_dev_module("R/mod_plop.R")
Error in file(file, ifelse(append, "a", "w")) : 
  cannot open the connection
In addition: Warning messages:
1: In file.create(where) :
  cannot create file '/Users/colin/Seafile/documents_colin/R/trash/ooooooo/dev/run_dev_R/mod_plop.R.R', reason 'No such file or directory'
2: In file(file, ifelse(append, "a", "w")) :
  cannot open file '/Users/colin/Seafile/documents_colin/R/trash/ooooooo/dev/run_dev_R/mod_plop.R.R': No such file or directory
```

+ Here is the output of the the file create by `golem::add_run_dev_module("mod_plop.R")`

``` r
# Run dev module mod_plop.R

options(golem.app.prod = FALSE)

golem::detach_all_attached()

golem::document_and_reload()


app_ui <- function(request) { 
     (id = 'run_dev')
}


app_server <- function(input, output, session) {
callModule(, 'run_dev')}


shinyApp(ui = app_ui, server = app_server)

```

There is something wrong with:

```
app_ui <- function(request) { 
     (id = 'run_dev')
}
```

Please also fix the indentation in 

``` r
app_server <- function(input, output, session) {
callModule(, 'run_dev')}
```

+ Please add a `cat_green_tick()` to confirm that the file has been created 

+ Please add at 

```
if ( open & rstudioapi::isAvailable() ) { 
      rstudioapi::navigateToFile(file = where)
    }
```

A message to open the file when not in RStudio, à la : 
https://github.com/ThinkR-open/golem/blob/dev/R/add_files.R#L190

Cheers :)

**Comment by Cervangirard** [comment]
Currently, with the new version of run_dev, we can't have a run_dev by module

**Comment by ColinFay** [comment]
I'm closing this for now

**Pull request closed without merging**